drivers/spi/spi_nxp_lpspi/spi_nxp_lpspi.c

```diff
@@ -235,10 +201,16 @@ static void lpspi_isr(const struct device *dev)
 		lpspi_handle_tx_irq(dev);
 	}
 
-	if (!spi_context_tx_on(ctx) && lpspi_is_rx_done(dev)) {
+	if (!spi_context_tx_on(ctx) && spi_context_rx_len_left(ctx) == 1) {
```

Comment:
`!spi_context_tx_on(ctx)` is checked up to 3 times. Move it out of the `if`s and add e.g.:
```
if (spi_context_tx_on(ctx)) {
	return;
}
```